encoding/marshal.go

```diff
@@ -1,91 +1,110 @@
-package encoding
-
-import (
-	"errors"
-	"reflect"
-)
-
-// A Marshaler can be encoded as a byte slice. (Note that Marshaler and
-// Unmarshaler are separate interfaces because Unmarshaler must have a pointer
-// receiver, while Marshaler does not.)
-type SiaMarshaler interface {
-	MarshalSia() []byte
-}
-
-// An Unmarshaler can be decoded from a byte slice. If a decoding error occurs,
-// UnmarshalSia should panic.
-type SiaUnmarshaler interface {
-	UnmarshalSia([]byte)
-}
-
-// Marshal encodes a value as a byte slice. The encoding rules are as follows:
+// Package encoding converts arbitrary objects into byte slices, and vis
+// versa. It also contains helper functions for reading and writing length-
+// prefixed data. The encoding rules are as follows:
```

Comment:
are you referring to a common practice, or a Sia-specific one? Because there's no encoding.go here, and I think it'd be silly to make a new file just for the docstring. It makes sense in marshal.go.